core/src/main/java/nl/nn/adapterframework/doc/model/ElementRole.java

```diff
@@ -16,26 +16,110 @@
 
 package nl.nn.adapterframework.doc.model;
 
+import java.util.Collection;
+import java.util.Comparator;
 import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
 
+import org.apache.logging.log4j.Logger;
+
+import lombok.AccessLevel;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
+import lombok.Setter;
 import nl.nn.adapterframework.doc.Utils;
+import nl.nn.adapterframework.util.LogUtil;
+
+public class ElementRole implements Comparable<ElementRole> {
+	private static Logger log = LogUtil.getLogger(ElementRole.class);
+
+	private static final Comparator<ElementRole> COMPARATOR =
+			Comparator.comparing(ElementRole::getSyntax1Name).thenComparing(role -> role.getElementType().getFullName());
 
-public class ElementRole {
 	private final @Getter ElementType elementType;
 	private final @Getter String syntax1Name;
 	private final int syntax1NameSeq;
+	private @Getter FrankElement defaultElementOptionConflict;
+	private Set<FrankElement> nameConflicts;
+	private Set<ElementRoleSet> participatesInRoleSets = new HashSet<>();
```

Comment:
Voor deze velden in een kort commentaar uitleggen/verwijzen waarvoor ze zijn

Reply:
Gedaan.